compio-io/src/read/buf.rs

```diff
@@ -96,7 +96,12 @@ impl<R: AsyncRead> AsyncBufRead for BufReader<R> {
         }
 
         if buf.need_fill() {
-            buf.with(|b| reader.read(b)).await?;
+            buf.with(|b| async move {
```

Comment:
Change `Inner` so that only available part is writable.

Reply:
No, we may also need the initialized part.